indico/web/client/js/jquery/widgets/jinja/principal_list_widget.js

```diff
@@ -1,28 +1,49 @@
-// This file is part of Indico.
-// Copyright (C) 2002 - 2020 CERN
-//
-// Indico is free software; you can redistribute it and/or
-// modify it under the terms of the MIT License; see the
-// LICENSE file for more details.
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {WTFPrincipalListField} from 'indico/react/components/WTFPrincipalListField';
 
 (function(global) {
   'use strict';
 
   global.setupPrincipalListWidget = function setupPrincipalListWidget(options) {
-    options = $.extend(
-      true,
-      {
+    options = {
+      ...{
         fieldId: null,
         eventId: null,
         openImmediately: false,
-        groups: null,
-        allowExternal: false,
-        allowNetworks: false,
+        withGroups: false,
+        withExternalUsers: false,
+        withEventRoles: false,
+        withCategoryRoles: false,
+        withRegistrationForms: false,
+        withNetworks: false,
         networks: [],
       },
-      options
+      ...options,
+    };
+    return options.legacy
+      ? _setupLegacyPrincipalListWidget(options)
+      : _setupPrincipalListWidget(options);
+  };
+
+  function _setupPrincipalListWidget(options) {
+    const field = document.getElementById(options.fieldId);
+    const principals = JSON.parse(field.value);
```

Comment:
Should this really be here and not inside the component like it's done for the date-related fields?

Reply:
Good question. Since this one doesn't really change and is only the initial value I thought we could avoid manipulating it through react. Having the manipulation in JS and the prop in react seems cleaner.